### PR TITLE
Only store completed scores in main scores table

### PR DESCRIPTION
**Pull request by nanaya**
The route is getting weird :thinking: 

This allows score processor or whatever to process completed scores sequentially.

- [x] lazer submitting fails and retries (for playcount)

### CODE REVIEW

File: app/Http/Controllers/Solo/ScoresController.php [hunk 1]
```diff
@@ -25,36 +26,13 @@ public function __construct()
...
-        return json_item($score, 'Solo\Score');
-    }
-
-    public function update($beatmapId, $id)
-    {
-        static::getBeatmapOrFail($beatmapId);
-
         $user = auth()->user();
-        $score = Score::where('user_id', $user->getKey())->findOrFail($id);
+        $scoreToken = ScoreToken::where([
```
Comment (peppy) [inline app/Http/Controllers/Solo/ScoresController.php:32]: Should probably have a check that `score_id` is `null`.

File: routes/web.php [hunk 2]
```diff
@@ -380,7 +380,10 @@
                 Route::get('scores', 'BeatmapsController@scores')->name('scores');
 
                 Route::group(['as' => 'solo.', 'prefix' => 'solo'], function () {
-                    Route::resource('scores', 'Solo\ScoresController', ['only' => ['store', 'update']]);
+                    Route::group(['namespace' => 'Solo'], function () {
+                        Route::post('scores', 'ScoreTokensController@store')->name('score-tokens.store');
+                        Route::put('scores/{token}', 'ScoresController@store')->name('scores.store');
```
Comment (notbakaneko) [inline routes/web.php:385]: shouldn't this be `store.update` 🤔

File: app/Http/Controllers/Solo/ScoresController.php [hunk 3]
```diff
@@ -5,74 +5,59 @@
...
-        } catch (PDOException $e) {
-            // TODO: move this to be a validation inside Score model
-            throw new InvariantException('failed updating score');
+        $scoreToken = ScoreToken::where([
+            'beatmap_id' => $beatmapId,
+            'user_id' => $user->getKey(),
+        ])->findOrFail($tokenId);
+
+        // return existing score otherwise (assuming duplicated submission)
+        if ($scoreToken->score_id === null) {
```
Comment (notbakaneko) [inline app/Http/Controllers/Solo/ScoresController.php:29]: `$scoreToken` should be locked for update if doing this

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: routes/web.php (hunk 2 above)
Comment (nanaya) [reply routes/web.php:385]: no because it's creating new score. The method stayed because that's what the client is using

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**